src/v2/content_digest.rs

```diff
@@ -0,0 +1,136 @@
+/// Implements types and methods for content verification
+use sha2::{self, Digest};
+use v2::*;
+
+/// ContentDigest stores a digest and its DigestAlgorithm
+#[derive(Clone, Debug, PartialEq)]
+pub struct ContentDigest {
+    digest: String,
+    algorithm: DigestAlgorithm,
+}
+
+/// DigestAlgorithm declares the supported algorithms
+#[derive(Display, Clone, Debug, PartialEq, EnumString)]
+pub enum DigestAlgorithm {
+    #[strum(to_string = "sha256")]
+    Sha256,
+}
+
+impl ContentDigest {
+    /// try_new attempts to parse the digest string and create a ContentDigest instance from it.error_chain
+    ///
+    /// Success depends on
+    /// - the string having a "algorithm:" prefix
+    /// - the algorithm being supported by DigestAlgorithm
```

Comment:
Do we want to add a "the digest is a valid value for DigestAlgorithm"? I think right now this accepts things like "sha256:thanks4allthefish".

Reply:
I thought about that too. We'd have one regular expression per algorithm to verify the digest further. I don't thing it's our responsibility though. If you strongly feel we should do it I'll incorporate it

Reply:
Not a strong feeling, I'm fine with keeping this as is for this PR.